x/multisig/types/genesis.go

```diff
@@ -25,6 +40,60 @@ func (m GenesisState) Validate() error {
 		return getValidateError(err)
 	}
 
+	keyIDs := make(map[exported.KeyID]bool, len(m.KeygenSessions)+len(m.Keys))
+	completedKeyIDs := make(map[exported.KeyID]bool)
+	for _, keygenSession := range m.KeygenSessions {
+		if keyIDs[keygenSession.GetKeyID()] {
+			return getValidateError(fmt.Errorf("duplicate key ID seen"))
+		}
+		keyIDs[keygenSession.GetKeyID()] = true
+
+		if err := keygenSession.ValidateBasic(); err != nil {
+			return getValidateError(err)
+		}
+	}
+	for _, key := range m.Keys {
```

Comment:
```suggestion

	for _, key := range m.Keys {
```

Reply:
I intentionally kept them together cuz those 2 blocks are both about keys and require uniquness of key IDs.